### DISCUSSION TITLE
Programmatically control isInteractive state value

### DISCUSSION THREAD

**Discussion by timseriakov** [xyflow/xyflow#3023] [Q&A] [answered]
Is there a way to programmatically control the value of `isInteractive` (state of the lock button is pressed)? 
and I'm also looking for a way to set the value of `isInteractive`: `false` **by default**. 
To switch to edit mode only after clicking the **custom UI Edit button**.

**Comment by timseriakov** [comment, marked as answer]
I found such a solution, perhaps someone will find it useful.
upd: [comment-5765086](https://github.com/wbkd/react-flow/discussions/3023#discussioncomment-5765086)

```js
import { ReactFlowState, useStore, useStoreApi } from 'reactflow'

const isInteractiveSelector = (s: ReactFlowState) => s.nodesDraggable && s.nodesConnectable && s.elementsSelectable

export const useCanvasInteractivity = () => {
  const store = useStoreApi()
  const isInteractiveState = useStore(isInteractiveSelector)

  const setInteractivity = (isInteractive: boolean) => {
    store.setState({
      nodesDraggable: isInteractive,
      nodesConnectable: isInteractive,
      elementsSelectable: isInteractive,
    })
  }

  const toggleInteractivity = () => setInteractivity(!isInteractiveState)

  return {
    isInteractive: isInteractiveState,
    setInteractivity,
    toggleInteractivity,
  }
}
```

**Reply by moklick** [reply]
thanks for sharing your solution! (You could define the selector outside of the hook)
